rust/src/dcerpc/parser.rs

```diff
@@ -49,216 +53,192 @@ fn assemble_uuid(uuid: Uuid) -> Vec<u8> {
-                seqnum: seqnum,
-                opnum: opnum,
-                ihint: ihint,
-                ahint: ahint,
-                fraglen: fraglen,
-                fragnum: fragnum,
-                auth_proto: auth_proto,
-                serial_lo: serial_lo,
-            }
-        )
-    )
-);
+pub fn parse_dcerpc_udp_header(i: &[u8]) -> IResult<&[u8], DCERPCHdrUdp> {
+    let (i, rpc_vers) = le_u8(i)?;
+    let (i, pkt_type) = le_u8(i)?;
+    let (i, flags1) = le_u8(i)?;
+    let (i, flags2) = le_u8(i)?;
+    let (i, drep) = take(3_usize)(i)?;
+    let endianness = if drep[0] == 0 { Endianness::Big } else { Endianness::Little };
+    let (i, serial_hi) = le_u8(i)?;
+    let (i, objectuuid) = take(16_usize)(i)?;
+    let (i, interfaceuuid) = take(16_usize)(i)?;
+    let (i, activityuuid) = take(16_usize)(i)?;
+    let (i, server_boot) = u32(endianness)(i)?;
+    let (i, if_vers) = u32(endianness)(i)?;
+    let (i, seqnum) = u32(endianness)(i)?;
+    let (i, opnum) = u16(endianness)(i)?;
+    let (i, ihint) = u16(endianness)(i)?;
+    let (i, ahint) = u16(endianness)(i)?;
+    let (i, fraglen) = u16(endianness)(i)?;
+    let (i, fragnum) = u16(endianness)(i)?;
+    let (i, auth_proto) = le_u8(i)?;
+    let (i, serial_lo) = le_u8(i)?;
+    let header = DCERPCHdrUdp {
+        rpc_vers,
+        pkt_type,
+        flags1,
+        flags2,
+        drep: drep.to_vec(),
+        serial_hi,
+        objectuuid: match parse_uuid(objectuuid) {
+            Ok((_, vect)) => assemble_uuid(vect),
+            Err(_e) => {
+                SCLogDebug!("{}", _e);
+                vec![0]
+            },
+        },
+        interfaceuuid: match parse_uuid(interfaceuuid) {
+            Ok((_, vect)) => assemble_uuid(vect),
+            Err(_e) => {
+                SCLogDebug!("{}", _e);
+                vec![0]
+            },
+        },
+        activityuuid: match parse_uuid(activityuuid){
+            Ok((_, vect)) => assemble_uuid(vect),
+            Err(_e) => {
+                SCLogDebug!("{}", _e);
+                vec![0]
+            },
+        },
+        server_boot,
+        if_vers,
+        seqnum,
+        opnum,
+        ihint,
+        ahint,
+        fraglen,
+        fragnum,
+        auth_proto,
+        serial_lo,
+    };
+    Ok((i, header))
+}
 
-named!(pub parse_dcerpc_bindack_result<DCERPCBindAckResult>,
-    do_parse!(
-        ack_result: le_u16
-        >>  ack_reason: le_u16
-        >>  transfer_syntax: take!(16)
-        >>  syntax_version: le_u32
-        >> (
-             DCERPCBindAckResult {
-                ack_result:ack_result,
-                ack_reason:ack_reason,
-                transfer_syntax:transfer_syntax.to_vec(),
-                syntax_version:syntax_version,
-            }
-        )
-    )
-);
+pub fn parse_dcerpc_bindack_result(i: &[u8]) -> IResult<&[u8], DCERPCBindAckResult> {
+    let (i, ack_result) = le_u16(i)?;
+    let (i,  ack_reason) = le_u16(i)?;
```

Comment:
nit: formatting is off

Reply:
I can fix this up on merge

Reply:
Oh, good catch. I had to disable auto-formatting to avoid creating huge mostly-indentation commits
Can you amend it?

Reply:
Forgot :(